pkg/controller/atlasproject/project.go

```diff
@@ -2,42 +2,171 @@ package atlasproject
+			r.Log.Info("Not removing Project from Atlas as per configuration")
+		} else {
+			if result := DeleteAllPrivateEndpoints(ctx, atlasProject.ID()); !result.IsOk() {
+				return r.terminate(ctx, workflow.ServerlessPrivateEndpointReady, errors.New(result.GetMessage()))
 			}
-			if p, _, err = ctx.Client.Projects.Create(ctx.Context, p, &mongodbatlas.CreateProjectOptions{}); err != nil {
-				return "", workflow.Terminate(workflow.ProjectNotCreatedInAtlas, err.Error())
+			if result := DeleteAllNetworkPeers(ctx.Context, atlasProject.ID(), ctx.SdkClient.NetworkPeeringApi, ctx.Log); !result.IsOk() {
+				return r.terminate(ctx, workflow.ProjectNetworkPeerIsNotReadyInAtlas, errors.New(result.GetMessage()))
+			}
+
+			err = r.syncAssignedTeams(ctx, atlasProject.ID(), atlasProject, nil)
+			if err != nil {
+				ctx.SetConditionFalse(api.ProjectTeamsReadyType)
+				return r.terminate(ctx, workflow.TeamNotCleaned, err)
+			}
+
+			if err = r.projectService.DeleteProject(ctx.Context, project.NewProject(atlasProject, orgID)); err != nil {
+				return r.terminate(ctx, workflow.Internal, err)
 			}
-			ctx.Log.Infow("Created Atlas Project", "name", project.Spec.Name, "id", p.ID)
-		} else {
-			return "", workflow.Terminate(workflow.ProjectNotCreatedInAtlas, err.Error())
 		}
+
+		if err = customresource.ManageFinalizer(ctx.Context, r.Client, atlasProject, customresource.UnsetFinalizer); err != nil {
+			return r.terminate(ctx, workflow.AtlasFinalizerNotRemoved, err)
+		}
+	}
+
+	return workflow.OK().ReconcileResult(), nil
+}
+
+func (r *AtlasProjectReconciler) ready(ctx *workflow.Context, projectID string) (ctrl.Result, error) {
+	ctx.EnsureStatusOption(status.AtlasProjectIDOption(projectID))
+	result := workflow.OK()
+	ctx.SetConditionFromResult(api.ProjectReadyType, result)
+	ctx.SetConditionFromResult(api.ReadyType, result)
+
+	return result.ReconcileResult(), nil
+}
+
+func (r *AtlasProjectReconciler) release(ctx *workflow.Context, atlasProject *akov2.AtlasProject) (ctrl.Result, error) {
+	if err := customresource.ManageFinalizer(ctx.Context, r.Client, atlasProject, customresource.UnsetFinalizer); err != nil {
+		return r.terminate(ctx, workflow.AtlasFinalizerNotRemoved, err)
+	}
+
+	return workflow.OK().ReconcileResult(), nil
+}
+
+func (r *AtlasProjectReconciler) manage(ctx *workflow.Context, atlasProject *akov2.AtlasProject, projectID string) (ctrl.Result, error) {
+	r.Log.Debugw("Add deletion finalizer", "name", customresource.FinalizerLabel)
+	if err := customresource.ManageFinalizer(ctx.Context, r.Client, atlasProject, customresource.SetFinalizer); err != nil {
+		return r.terminate(ctx, workflow.AtlasFinalizerNotSet, err)
 	}
 
-	if p == nil || p.ID == "" {
-		ctx.Log.Error("Project or its project ID are empty")
-		return "", workflow.Terminate(workflow.Internal, "")
+	ctx.EnsureStatusOption(status.AtlasProjectIDOption(projectID))
+	result := workflow.InProgress(workflow.ProjectBeingConfiguredInAtlas, "configuring project in Atlas")
```

Comment:
question: Is this a net-new status condition or a change?

Reply:
It's a new one, not breaking change